docs/DeveloperGuide.md

```diff
@@ -221,7 +222,65 @@ for subsequent prompts.
+or symbols that are deemed as illegal (see [Illegal Characters & Phrases](#illegal-characterse--phrases) for details).
+If at least one illegal character or phrase is found, an `InvalidCommandException` will be thrown and the parsing is
+aborted.
+
+**(Steps 4 to 10)** If no illegal characters and phrases are found, `Parser#parseUserInput()` will examine the first
+word in the user input. This first word should represent the command type that the user wish to execute (i.e. `exercise`,
+`workout`, `plan`, `schedule`, `search`, `help`, or `exit`). Depending on the first word of the user input, different
+methods will be invoked to create the appropriate object of the subclass of the `Command` abstract superclass (see
+example).
+However, if the first word is not a valid command type, an `InvalidCommandException` will be thrown and the parsing is
+aborted.
+- For example, if the user input is `workout /new push up /reps 10`, `Parser#createWorkoutCommand()` will be invoked
+and a `WorkoutCommand` object will be returned by this method.
+
+Inside each of these 'create command' methods, the following generalised procedure to create an object of the subclass 
+of `Command` is carried out:
+1. (For commands that expect an action keyword (e.g. `/list`, `/new`)) The action keyword is parsed and determined.
+    - If the action keyword is invalid, an `InvalidCommandException` is thrown and the parsing is aborted.
+2. Depending on the action keyword (or lack thereof), the number of arguments are checked.
+    - If insufficient or too many arguments are provided in the user input, an `InvalidCommandException` is thrown
+   and the parsing is aborted.
+3. A new object of the subclass of `Command` is created and if the object is successfully constructed with no errors,
+it is returned to `Parser#parseUserInput()`.
+
+**(Step 11)** The object created is then returned to `WerkIt#startContinuousUserInput()`.
+
+<span class="box info"> 🧾 (About the sequence diagram) Stricly speaking, the object is returned right after whichever 
+'create command' method is invoked. However, to improve the readability of the diagram, only one return line is shown,
+since all alternate paths will return an object that is a subclass of the `Command` class.</span>
+
+**(Step 12)** The final step of this section is to invoke the `Command#execute()` method, which will in turn call the
+overridden `execute()` method of the subclass of `Command`.
+- For example, if the user input is `workout /new push up /reps 10`, the created `WorkoutCommand` object is upcasted
+to `Command` when returned to `WerkIt#startContinuousUserInput()`, but when `newCommand.execute()` is called,
+`WorkoutCommand#execute()` is called.
+
+Thereafter, the appropriate procedures are taken to complete the task requested by the user. The various procedures
+are explained in later sections of this developer guide.
+
+#### Illegal Characterse & Phrases
```

Comment:
Line 275: Typo on the word "Characters"

Reply:
Thanks for spotting the errata. It has been fixed in Commit dd5ca9b